crates/matrix-sdk-crypto/src/machine.rs

```diff
@@ -862,24 +861,26 @@ impl OlmMachine {
         #[derive(Deserialize)]
         struct ContentStub<'a> {
             #[serde(borrow, rename = "org.matrix.msgid")]
-            message_id: JsOption<&'a str>,
+            message_id: Option<&'a str>,
         }
 
         #[derive(Deserialize)]
         struct ToDeviceStub<'a> {
+            sender: &'a str,
+            #[serde(rename = "type")]
+            event_type: &'a str,
             #[serde(borrow)]
             content: ContentStub<'a>,
         }
 
-        let message_id = event
-            .deserialize_as::<ToDeviceStub<'_>>()
-            .ok()
-            .and_then(|t| t.content.message_id.into_option());
-
-        tracing::Span::current().record("message_id", message_id);
+        if let Ok(event) = event.deserialize_as::<ToDeviceStub<'_>>() {
+            tracing::Span::current().record("sender", event.sender);
+            tracing::Span::current().record("event_type", event.event_type);
+            tracing::Span::current().record("message_id", event.content.message_id);
```

Comment:
```suggestion
            if let Some(message_id) = event.content.message_id {
                tracing::Span::current().record("message_id", message_id);
            }
```

Reply:
Hmm, I thought that I need this, but the `None` isn't displayed when I record it unconditionally.